Assets/HoloToolkit/SpatialSound/Scripts/AudioInfluencerManager.cs

```diff
@@ -0,0 +1,242 @@
+        /// Frequency above the nominal range of human hearing.
+        /// </summary>
+        /// <remarks>
+        /// This frequency can be used to set a low pass filter to allow all
+        /// audible frequencies through the filter.
+        /// </remarks>
+        public static readonly float NeutralHighFrequency = 22000.0f;
+
+        /// <summary>
+        /// Time, in seconds, between audio influence updates.
+        /// </summary>
+        /// <remarks>
+        /// The UpdateInterval range is betweel 0.0 and 1.0, inclusive.
+        /// The default value is 0.25.
+        /// A value of 0.0f indicates that updates occur every frame.
+        /// </remarks>
+        [Tooltip("Time, in seconds, between audio influence updates.  0 indicates to update every frame.")]
+        [Range(0.0f, 1.0f)]
+        public float UpdateInterval = 0.25f;
+
+        /// <summary>
+        /// Maximum distance, in meters, to look when attempting to find the user and any influencers.
+        /// </summary>
+        /// <remarks>
+        /// The MaxDistance range is 1.0 to 50.0, inclusive.
+        /// The default value is 20.0.
+        /// </remarks>
+        [Tooltip("Maximum distance, in meters, to look when attempting to find the user and any influencers.")]
+        [Range(1.0f, 50.0f)]
+        public float MaxDistance = 20.0f;
+
+        /// <summary>
+        /// Maximum number of objects that will be considered when looking for influencers. 
+        /// Setting this value too high may have a negative impact on the performance of your experience.
+        /// </summary>
+        /// <remarks>
+        /// MaxObjects can only be set in the Unity Inspector.
+        /// The MaxObjects range is 1 to 25, inclusive.
+        /// The default value is 10.
+        /// </remarks>
+        [Tooltip("Maximum number of objects that will be considered when looking for influencers.")]
+        [Range(1, 25)]
+        [SerializeField]
+        private int MaxObjects = 10;
+
+        // Time of last audio processing update.
+        private DateTime lastUpdate = DateTime.MinValue;
+
+        // The source of the audio.
+        private AudioSource audioSource;
+
+        // The initial volume level of the audio source.
+        private float initialAudioSourceVolume;
+
+        // The hits returned by Physics.RaycastAll
+        private RaycastHit[] hits;
+
+        // The collection of previously applied audio influencers.
+        private List<IAudioInfluencer> previousInfluencers = new List<IAudioInfluencer>();
+
+        // Potential effects manipulated by an audio influencer and their key
+        // properties.
+        private AudioLowPassFilter lowPassFilter;
+        private float nativeLowPassCutoffFrequency;
+        private AudioHighPassFilter highPassFilter;
+        private float nativeHighPassCutoffFrequency;
+
+        private void Awake() 
+        {           
+            audioSource = gameObject.GetComponent<AudioSource>();
```

Comment:
let's check for null here in case the dev set it in the inspector

Reply:
this comes from querying the game object. this class, by design, does not expose the AudioSource to the inspector. this was true of the earlier version as well.